turbopack/crates/turbo-tasks-testing/tests/detached.rs

```diff
@@ -2,53 +2,140 @@
 
 use tokio::{
     sync::{watch, Notify},
-    time::{timeout, Duration},
+    time::{sleep, timeout, Duration},
 };
-use turbo_tasks::{turbo_tasks, Completion, TransientInstance, Vc};
+use turbo_tasks::{turbo_tasks, State, TransientInstance, Vc};
 use turbo_tasks_testing::{register, run, Registration};
 
 static REGISTRATION: Registration = register!();
 
 #[tokio::test]
 async fn test_spawns_detached() -> anyhow::Result<()> {
     run(&REGISTRATION, || async {
-        let notify = TransientInstance::new(Notify::new());
-        let (tx, mut rx) = watch::channel(None);
+        timeout(Duration::from_secs(5), async {
```

Comment:
Is this timeout just to avoid hanging / deadlocks?

Reply:
Yes, I can add a comment.